server/channels/app/channel.go

```diff
@@ -3812,3 +3812,20 @@ func (s *Server) getDirectChannel(c request.CTX, userID, otherUserID string) (*m
 
 	return channel, nil
 }
+
+func (a *App) CheckIfChannelIsRestrictedDM(c request.CTX, channel *model.Channel) (bool, *model.AppError) {
+	if *a.Config().TeamSettings.RestrictDirectMessage != model.DirectMessageTeam {
```

Comment:
not- If we check for `model.DirectMessageAny`, it will be slight more future proof

Reply:
Actually I think we only want this logic to run when the setting is on "Team", because if we were to add another setting to restrict this, the logic would likely be completely different.

Reply:
Ah okay, got it.